data-processing/common/parse_equation.py

```diff
@@ -64,23 +66,54 @@ def get_symbols(mathml: str) -> List[Symbol]:
             if index is not None:
                 characters.append(index)
 
-        # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
-        # annotations are available for processing later nodes.
-        node_clone = copy.copy(symbol_node)
-        _remove_s2_annotations(node_clone)
-        for descendant in node_clone.descendants:
-            _remove_s2_annotations(descendant)
+        # Group consecutive character symbols only if they are a child of the mrow tag.
+        # The .has_attr('s2:end') is necessary as not every symbol is annotated by katex
+        if symbol_node.has_attr('s2:end') and symbol_node.parent.name in VALID_PARENT_TAG and symbol_node.name in CHARACTER_TAGS:
```

Comment:
nit pick: Is there a more descriptive variable name than `VALID_PARENT_TAGS`, something that conveys that it's tags for parents of clusters of characters or spans of mergeable characters?